File: src/api/sendEmail.mjs
import express from 'express';
import cors from 'cors';
import fetch from 'node-fetch';
import dotenv from 'dotenv';
dotenv.config();

const app = express();
app.use(cors());
app.use(express.json());

app.post('/api/send-email', async (req, res) => {
  const { to, name } = req.body;
  const apiKey = process.env.REACT_APP_RESEND_API_KEY;
  if (!apiKey) return res.status(500).json({ error: 'Resend API key missing' });

  try {
    const response = await fetch('https://api.resend.com/emails', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        from: 'IEC Registration <ecell@studentvault.xyz>', // Use a verified sender
        to,
        subject: 'IEC Registration Successful',
        html: `<p>Hi ${name},<br/>Thank you for registering! Here are the details of your registration:</p>`
      })
    });
    const result = await response.json();
    if (!response.ok) {
      return res.status(400).json({ error: result });
    }
    res.json({ success: true, result });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

const PORT = process.env.PORT || 5000;
app.listen(PORT, () => {
  console.log(`Email server running on port ${PORT}`);
});

File: src/api/send_email.py
from flask import Flask, request, jsonify
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart

app = Flask(__name__)

SENDER_EMAIL = "your_email@gmail.com"  # Replace with your email
SENDER_PASSWORD = "your_app_password"  # Replace with your app password

@app.route('/api/send-email', methods=['POST'])
def send_email():
    data = request.get_json()
    to = data.get('to')
    name = data.get('name')
    subject = "Registration Confirmation"
    body = f"Hello {name},\n\nThank you for registering!"

    msg = MIMEMultipart()
    msg['From'] = SENDER_EMAIL
    msg['To'] = to
    msg['Subject'] = subject
    msg.attach(MIMEText(body, 'plain'))

    try:
        with smtplib.SMTP_SSL('smtp.gmail.com', 465) as server:
            server.login(SENDER_EMAIL, SENDER_PASSWORD)
            server.sendmail(SENDER_EMAIL, to, msg.as_string())
        return jsonify({"success": True}), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 500

if __name__ == '__main__':
    app.run(port=5000)
